src/ValidatesOpenApiSpec.php

```diff
@@ -226,10 +226,14 @@ protected function getSpecFileType(): string
      */
     protected function getAuthenticatedRequest(SymfonyRequest $request): SymfonyRequest
     {
-        $authenticatedRequest = clone $request;
-        $authenticatedRequest->headers->set('Authorization', 'Bearer token');
-
-        return $authenticatedRequest;
+        // Spoofing when authentication headers are not present.
+        if ($request->headers->has('Authorization')) {
+            return $request;
+        } else {
+            $authenticatedRequest = clone $request;
+            $authenticatedRequest->headers->set('Authorization', 'Bearer token');
+            return $authenticatedRequest;
+        }
```

Comment:
nit: a little code style change--no need for the `else` since we're returning.
```suggestion
        if ($request->headers->has('Authorization')) {
            return $request;
        }

        $authenticatedRequest = clone $request;
        $authenticatedRequest->headers->set('Authorization', 'Bearer token');
        
        return $authenticatedRequest;
```